packages/core/aqua-to-js/src/generate/validators.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ArrowWithoutCallbacks,
+  FunctionCallDef,
+  JSONValue,
+  SimpleTypes,
+  UnlabeledProductType,
+} from "@fluencelabs/interfaces";
+
+import { typeToTs } from "../common.js";
+
+export function validateFunctionCall(
+  schema: FunctionCallDef,
+  ...args: JSONValue[]
+) {
+  const schemaArgs =
+    schema.arrow.domain.tag === "nil"
+      ? []
+      : Object.values(schema.arrow.domain.fields);
+
+  if (args.length !== schemaArgs.length) {
+    throw new Error(
+      `Expected ${schemaArgs.length} arguments but provided ${args.length}`,
+    );
+  }
+
+  for (let i = 0; i < args.length; i++) {
+    validateFunctionCallArg(schemaArgs[i], args[i], i + 1);
```

Comment:
some doc on `i+1` would be nice. Not clear why you do that.

Reply:
Changed parameter name from `argIndex` to `argPosition`. Should be clearer now